### ISSUE TITLE
Next iteration of the docs

### ISSUE THREAD

**Issue by adrianthedev** [avo-hq/avo#2551]
The docs have evolved quite a bit, and we need to think of a way to refactor/reorganize them.

Let's use this issue to collect all the feedback we received.

- https://github.com/avo-hq/avo/discussions/1144#discussioncomment-8645974
- add an `avo.rb` page where we talk about each feature from the initializer.
- https://github.com/avo-hq/avodocs/issues/195

**Comment by Paul-Bob** [comment]
Sometimes syntax change several times, on the docs it's hard to follow all the warnings: use this on version x and that on y or use this on last version. We should somehow manage the docs according to the selected version. Let's find a way to do that without duplicate all docs pages for each version.

**Comment by sedubois** [comment]
The `use_resource` section of https://docs.avohq.io/3.0/field-options.html is empty

**Comment by sedubois** [comment]
The avohq.io landing page is still linking to Avo 2 docs. In the section "All the tools you need to build your new app fast", e.g. click "Fantastic CRUD UI", it goes to https://docs.avohq.io/2.0/resources.html.

**Comment by sedubois** [comment]
The avohq.io "Full mobile interface" and "Multi-tenancy" links under "All the tools you need to build your new app fast" just link to the current page, avohq.io.

**Comment by sedubois** [comment]
- Broken video play button placement on mobile
- consider using center alignment for CTA

| | |
| - | - |
| ![Screen Shot 2024-03-04 at 11 03 18](https://github.com/avo-hq/avo/assets/4217871/84913f17-40e9-4c5c-8186-04dba34c9498) | ![Screen Shot 2024-03-04 at 11 03 59](https://github.com/avo-hq/avo/assets/4217871/046f6fd6-bfac-457c-9682-6f3995c85ac9) |

**Comment by Paul-Bob** [comment]
> The avohq.io "Full mobile interface" and "Multi-tenancy" links under "All the tools you need to build your new app fast" just link to the current page, avohq.io.

Thank you, fixed

**Comment by sedubois** [comment]
I recommend providing visual highlights of Avo's main features somewhere prominent, such as on avohq.io in the features section ("All the tools you need to build your new app fast") and in another equivalent place in the beginning of the documentation. 

Currently there is a grid with feature names such as "Actions", "Filters", etc but it would be more visually compelling if this was changed to show images/GIFs too. Such GIFs are already available within the individual doc pages, so they could just be collected together.

Alongside (or alternately to) the visuals, there could also be code snippets.

So each feature card would have a few essential pieces of information: 
- title, description and anything else marketing-friendly
- visual (still image or animation)
- code snippet
- link to learn more

**Comment by sedubois** [comment]
There could be a warning when viewing the old v2 documentation, with a link to go to the v3.

For example when viewing this:
https://docs.avohq.io/2.0/custom-fields.html

There could be a block at the top saying "You are viewing the documentation for Avo 2.0 which is in maintenance. [Click here](https://docs.avohq.io/3.0/custom-fields.html) to view the Avo 3.0 documentation."

BTW such v2 documentation comes up in search engines instead of the v3 for some reason. Perhaps something to be tweaked in the website's robots/SEO/meta tags config?

**Comment by sedubois** [comment]
[As mentioned in Discord](https://discord.com/channels/740892036978442260/1214265077209759805/1214480082685399080), the search feature could be improved.

**Comment by adrianthedev** [comment]
> [As mentioned in Discord](https://discord.com/channels/740892036978442260/1214265077209759805/1214480082685399080), the search feature could be improved.

What would you like to see improved? The user interface, the searching, matching, speed?

We use [Algolia doc search](https://docsearch.algolia.com/) which comes baked in with vitepress and needs less effort from our part.
We'd like to consolidate our efforts to building the framework, but will take in any improvements from the community. The [docs repo](https://github.com/avo-hq/avodocs) is open source.

**Comment by sedubois** [comment]
> the search feature could be improved

> What would you like to see improved?

@adrianthedev let's take the example of finding how to scope items in a association field. As I want to learn how to configure that, I go to the documentation and will want to search for an option that applies to the `field` DSL, so I went to the [field options documentation page](https://docs.avohq.io/3.0/field-options.html). The very name of that page makes it clear that whatever the `field` DSL supports as options should be listed on that page. However I went through that whole page and found nothing about scoping the field to only include specific records. I also hit Cmd-F on the keyboard and searched for "scop" but nothing relevant came up either. The page should list and hyperlink to options that are described elsewhere in the docs (in this case, the option I was looking for was "attach_scope").

Also broken link in the docs: 

https://docs.avohq.io/3.0/search.html#enable-search-for-a-resource

> In this block, you may configure the search however strict or loose you need it. Check out [ransack's search matchers](https://github.com/activerecord-hackery/ransack#search-matchers) to compose the query better.

The Ransack README has no section with hash `#search-matchers`.

**Comment by sedubois** [comment]
Actions documentation: 
https://docs.avohq.io/3.0/actions.html

the code syntax could be simplified with the one suggested by Paul:
https://discord.com/channels/740892036978442260/1222107435351805973/1222115078187253780

Before:
```rb
def handle(**args)
  query, fields, current_user, resource = args.values_at(:query, :fields, :current_user, :resource)
  # ...
end
```

After:
```rb
def handle(query:, fields:, current_user:, resource:, **args)
  # ...
end
```

**Comment by sedubois** [comment]
> Would you be able to open a PR with that suggestion? https://github.com/avo-hq/avodocs

FYI @adrianthedev this is waiting for a review: 
https://github.com/avo-hq/avodocs/pull/184

**Comment by Paul-Bob** [comment]
Thank you @sedubois, we'll have a look

**Comment by sedubois** [comment]
The map documentation mentions "you need to add the `map_view` class attribute" but the code snippet below then uses `self.map`.

https://docs.avohq.io/3.0/map-view.html
